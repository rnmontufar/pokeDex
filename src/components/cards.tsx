import { useHistory } from "react-router-dom";
import { getColor } from "../helpers/helpers";

export function Cards(props: any) {
  let { pokemons } = props;
  let history = useHistory();

  function handleClick(pokemon: any) {
    history.push({
      pathname: '/details',
      state: { pokemon: pokemon }
    });
  }

  return (
    <div className="bg-gradient-to-r from-green-100 via-red-300 to-yellow-600 mt-6">
      <div className="flex flex-wrap items-stretch justify-center m-auto max-w-6xl">
        {pokemons.map((pokemon: any) => (
          <div className={`bg-${getColor(pokemon.types[0].type.name)}-200 rounded-xl m-2 p-3 shadow-2xl hover:bg-opacity-20 cursor-pointer`} onClick={()=>handleClick(pokemon)}>
          <div className="bg-gray-100 h-36 w-36 rounded-full mx-auto">
              <img className="h-15 w-15 m-auto h-36 w-36" src={pokemon.sprites.front_default} alt={pokemon.name} />
            </div>
            <div className="mt-4">
              <span className="text-gray-500 text-xs rounded-xl px-2 bg-yellow-200">{'#' + pokemon.id.toString().padStart(3, '0')}</span>
              <div className="text-gray-900 text-lg" >{pokemon.name[0].toUpperCase() + pokemon.name.slice(1)}</div>
            </div>
            <div className="mt-1">
              {pokemon.types.map((type: any) => (
                <span key={type.type.name} className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-100 text-blue-800 mr-1">
                  {type.type.name[0].toUpperCase() + type.type.name.slice(1)}
                </span>
              ))}
            </div>

            <div className="mt-1">
              {pokemon.abilities.map((abilitie: any) => (
                <span key={abilitie.ability.name} className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-300 text-blue-800 mr-1">
                  {abilitie.ability.name[0].toUpperCase() + abilitie.ability.name.slice(1)}
                </span>
              ))}
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}