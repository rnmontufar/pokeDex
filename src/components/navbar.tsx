import pokeDex from "../pokedex.png"; 

export function Navbar() {
  return (
    <div className="bg-gray-200 transform translate-x-10 translate-y-4 bg-gradient-to-br from-blue-400 to-indigo-500">
      <div className="flex flex-wrap justify-center text-6xl">
        <span >PokeDex</span>
        <img src={pokeDex} alt='pokedex' className="w-21 h-14 m-2" />
      </div>
      <div className="flex flex-wrap justify-center text-xl">
        Kanto edition
      </div>
    </div>
  )
}
