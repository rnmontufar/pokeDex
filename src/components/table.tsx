import { useHistory } from "react-router-dom";

/* eslint-disable jsx-a11y/alt-text */
export function Table(props: any) {
  console.log(props.pokemons);
  let { pokemons } = props;
  let history = useHistory();

  function handleClick(pokemon: any) {
    history.push({
      pathname: '/details',
      state: { pokemon: pokemon }
    });
  }

  const colors: { [key: string]: any } = {
    fire: 'red-100',
    grass: 'green-100',
    electric: 'yellow-100',
    water: 'blue-200',
    ground: 'yellow-100',
    rock: 'gray-200',
    fairy: 'blue-200',
    poison: 'gray-200',
    bug: 'gray-300',
    dragon: 'red-200',
    psychic: 'pink-200',
    flying: 'yellow-200',
    fighting: 'blue-300',
    normal: 'pink-100',
    ice: 'gray-100'
  };

  const getColor = (name: any) => {
    return colors[name];
  };

  return (
    <div className="bg-gradient-to-r from-green-100 via-red-300 to-yellow-600 mt-6">
      <div className="flex flex-wrap items-stretch justify-center m-auto max-w-6xl">
        {pokemons.map((pokemon: any) => (
          <div className={`bg-${getColor(pokemon.types[0].type.name)} rounded-xl m-2 p-3 shadow-2xl hover:bg-opacity-20 cursor-pointer`} onClick={()=>handleClick(pokemon)}>
          <div className="bg-gray-100 h-36 w-36 rounded-full mx-auto">
              <img className="h-15 w-15 m-auto h-36 w-36" src={pokemon.sprites.front_default} alt={pokemon.name} />
            </div>
            <div className="mt-4">
              <span className="text-gray-500 text-xs rounded-xl px-2 bg-yellow-200">{'#' + pokemon.id.toString().padStart(3, '0')}</span>
              <div className="text-gray-900 text-lg" >{pokemon.name[0].toUpperCase() + pokemon.name.slice(1)}</div>
            </div>
            <div className="mt-1">
              {pokemon.types.map((type: any) => (
                <span key={type.type.name} className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-100 text-blue-800 mr-1">
                  {type.type.name[0].toUpperCase() + type.type.name.slice(1)}
                </span>
              ))}
            </div>

            <div className="mt-1">
              {pokemon.abilities.map((abilitie: any) => (
                <span key={abilitie.ability.name} className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-300 text-blue-800 mr-1">
                  {abilitie.ability.name[0].toUpperCase() + abilitie.ability.name.slice(1)}
                </span>
              ))}
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}