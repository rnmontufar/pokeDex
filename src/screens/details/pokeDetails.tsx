
import { colors200, colors400 } from "../../helpers/helpers";

export function PokeDetails(props: any) {
  const pokemon = props.location.state.pokemon;
  return (
    <div className="mt-6 ">
      <div className="flex flex-wrap items-stretch justify-center m-auto max-w-xl">
        <figure className={`${colors400[pokemon.types[0].type.name]} bg-opacity-70 m-2 p-3 shadow-2xl hover:bg-opacity-90 bg-gray-100 rounded-xl p-8">`}>
          <div className="bg-gray-100 h-36 w-36 rounded-full mx-auto">
            <img className="w-32 h-32 rounded-full mx-auto" src={pokemon.sprites.front_default} alt={pokemon.name} width="384" height="512" />
            <span className="text-gray-500 text-xs rounded-xl px-2 bg-yellow-200">{'#' + pokemon.id.toString().padStart(3, '0')}</span>
            <div className="text-gray-900 text-lg text-center" >{pokemon.name[0].toUpperCase() + pokemon.name.slice(1)}</div>
          </div>
          <div>
            <blockquote>
              <div className="mt-6">
                <span className="text-sm mr-1">Weight:</span>
                <span key={pokemon.weight} className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${colors200[pokemon.types[0].type.name]} text-blue-900 mr-1`}>
                  {pokemon.weight}
                </span>
                <div className="mt-1">
                  <span className="text-sm mr-1">Height:</span>
                  <span key={pokemon.height} className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${colors200[pokemon.types[0].type.name]} text-blue-900 mr-1`}>
                    {pokemon.height}
                  </span>
                </div>
                <div className="mt-1">
                  <span className="text-sm mr-1">Type:</span>
                  {pokemon.types.map((type: any) => (
                    <span key={type.type.name} className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${colors200[pokemon.types[0].type.name]} text-blue-900 mr-1`}>
                      {type.type.name[0].toUpperCase() + type.type.name.slice(1)}
                    </span>
                  ))}
                </div>
                <div className="mt-1">
                  <span className="text-sm mr-1">Abilities:</span>
                  {pokemon.abilities.map((abilitie: any) => (
                    <span key={abilitie.ability.name} className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${colors200[pokemon.types[0].type.name]} text-gray-900 mr-1`}>
                      {abilitie.ability.name[0].toUpperCase() + abilitie.ability.name.slice(1)}
                    </span>
                  ))}
                </div>
                <div className="mt-1">
                  <span className="text-sm mr-1">Moves:</span>
                  {pokemon.moves.map((move: any) => (
                    <span key={move.move.name} className={`px-2 inline-flex text-xs leading-5 font-semibold rounded-full ${colors200[pokemon.types[0].type.name]} text-gray-900 mr-1`}>
                      {move.move.name[0].toUpperCase() + move.move.name.slice(1)}
                    </span>
                  ))}
                </div>
              </div>
            </blockquote>
          </div>
        </figure>
      </div>
    </div>
  )
}
