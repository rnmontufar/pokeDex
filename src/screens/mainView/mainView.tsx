import { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useSelector } from "react-redux";  
import { Cards } from "../../components/cards";
import { getPokemons } from "./redux/actions";
import { getPokemonsResult, isLoadingGetPokemonsResult } from "./redux/selectors";

export function MainView() {
  const dispatch = useDispatch();
  const pokemons = useSelector(getPokemonsResult)
  const isLoadingpokemonsReponse = useSelector(isLoadingGetPokemonsResult)
  const [initialState, setInitialState] = useState(false);

  useEffect(() => {
    setInitialState(true);
    dispatch(getPokemons({payload: "pikachu" }));
  }, [dispatch]);

  return (
    <div>
      <header >
        {!isLoadingpokemonsReponse && initialState ? <Cards pokemons={pokemons}/> : "Loading.."}
      </header>
    </div>
  )
}
