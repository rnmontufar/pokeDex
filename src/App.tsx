import { Provider } from "react-redux";
import { MainView } from "./screens/mainView/mainView";
import { PokeDetails } from "./screens/details/pokeDetails";
import { store } from "./redux/store";
import { Route, BrowserRouter as Router, Switch } from "react-router-dom";
import { Navbar } from "./components/navbar";
import './index.css';

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Navbar/>
        <Switch>
          <Route path="/details" exact component={PokeDetails} />
          <Route path="/" exact component={MainView} />
        </Switch>
      </Router>
    </Provider>
  );
}

export default App;