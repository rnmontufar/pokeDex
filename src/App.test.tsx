import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import { MainView } from "./screens/mainView/mainView";
import { Cards } from "../src/components/cards";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { PokeDetails } from './screens/details/pokeDetails';

const initialState = {
  getPokemons: {
    getPokemonsResult: [
      {
        abilities: [{
          ability: {
            name: "overgrow",
          }
        }],
        id: 1,
        name: "bulbasaur",
        types: [{
          type: {
            name: "grass",
          }
        }],
        sprites: {
          front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png"
        }
      }
    ]
  }
};

const location = {
  state:{
    pokemon:{
      abilities: [{
        ability: {
          name: "overgrow",
        }
      }],
      id: 1,
      moves: [{
        move: {
          name: "razor-wind",
        }
      }],
      name: "bulbasaur",
      sprites: {
        front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png"
      },
      types: [{
        type: {
          name: "grass",
        }
      }]
    }
  }
}

afterEach(() => {
  cleanup();
});

test('Render MainView', () => {
  const mockStore = configureStore();
  const store = mockStore(initialState)
  const tree = renderer.create(<Provider store={store}><MainView /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Cards', () => {
  const tree = renderer.create(<Cards pokemons={initialState.getPokemons.getPokemonsResult} />).toJSON();
  expect(tree).toMatchSnapshot();
})

test('Render PokeDetails', () => {
  const tree = renderer.create(<PokeDetails location={location} />).toJSON();
  expect(tree).toMatchSnapshot();
})