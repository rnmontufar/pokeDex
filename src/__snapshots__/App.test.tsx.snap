// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cards 1`] = `
<div
  className="bg-gradient-to-r from-green-100 via-red-300 to-yellow-600 mt-6"
>
  <div
    className="flex flex-wrap items-stretch justify-center m-auto max-w-6xl"
  >
    <div
      className="bg-green-200 bg-opacity-40 rounded-xl m-2 p-3 shadow-2xl hover:bg-opacity-20 cursor-pointer"
      onClick={[Function]}
    >
      <div
        className="bg-gray-100 h-36 w-36 rounded-full mx-auto"
      >
        <img
          alt="bulbasaur"
          className="h-15 w-15 m-auto h-36 w-36"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png"
        />
      </div>
      <div
        className="mt-4"
      >
        <span
          className="text-gray-500 text-xs rounded-xl px-2 bg-yellow-200"
        >
          #001
        </span>
        <div
          className="text-gray-900 text-lg"
        >
          Bulbasaur
        </div>
      </div>
      <div
        className="mt-1"
      >
        <span
          className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-100 text-blue-800 mr-1"
        >
          Grass
        </span>
      </div>
      <div
        className="mt-1"
      >
        <span
          className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-blue-300 text-blue-800 mr-1"
        >
          Overgrow
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render MainView 1`] = `
<div>
  <header>
    <div
      className="flex justify-center m-auto mt-3 transform translate-x-10 translate-y-4 text-base"
    >
      Loading...
    </div>
  </header>
</div>
`;

exports[`Render PokeDetails 1`] = `
<div
  className="mt-6 "
>
  <div
    className="flex flex-wrap items-stretch justify-center m-auto max-w-xl"
  >
    <figure
      className="bg-green-400 bg-opacity-70 m-2 p-3 shadow-2xl hover:bg-opacity-90 bg-gray-100 rounded-xl p-8\\">"
    >
      <div
        className="bg-gray-100 h-36 w-36 rounded-full mx-auto"
      >
        <img
          alt="bulbasaur"
          className="w-32 h-32 rounded-full mx-auto"
          height="512"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png"
          width="384"
        />
        <span
          className="text-gray-500 text-xs rounded-xl px-2 bg-yellow-200"
        >
          #001
        </span>
        <div
          className="text-gray-900 text-lg text-center"
        >
          Bulbasaur
        </div>
      </div>
      <div>
        <blockquote>
          <div
            className="mt-6"
          >
            <span
              className="text-sm mr-1"
            >
              Weight:
            </span>
            <span
              className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-blue-900 mr-1"
            />
            <div
              className="mt-1"
            >
              <span
                className="text-sm mr-1"
              >
                Height:
              </span>
              <span
                className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-blue-900 mr-1"
              />
            </div>
            <div
              className="mt-1"
            >
              <span
                className="text-sm mr-1"
              >
                Type:
              </span>
              <span
                className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-blue-900 mr-1"
              >
                Grass
              </span>
            </div>
            <div
              className="mt-1"
            >
              <span
                className="text-sm mr-1"
              >
                Abilities:
              </span>
              <span
                className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-gray-900 mr-1"
              >
                Overgrow
              </span>
            </div>
            <div
              className="mt-1"
            >
              <span
                className="text-sm mr-1"
              >
                Moves:
              </span>
              <span
                className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-gray-900 mr-1"
              >
                Razor-wind
              </span>
            </div>
          </div>
        </blockquote>
      </div>
    </figure>
  </div>
</div>
`;
